tests/foreman/api/test_hostgroup.py

```diff
@@ -1,9 +1,175 @@
+            if result['name'] == 'ntp':
+                content_view.add_puppet_module(
+                    result['author'],
+                    result['name']
+                )
+                break
+
+        content_view.publish()
+        cvv = entities.ContentViewVersion(
+            id=content_view.read_json()['versions'][0]['id']
+        )
+        lc_env = entities.LifecycleEnvironment(
+            organization=org,
+            name=gen_string('alpha')
+        ).create()
+        cvv.promote(lc_env.id)
+        cv_attrs = content_view.read_json()
+        self.assertEqual(len(cv_attrs['versions']), 1)
+        self.assertEqual(len(cv_attrs['puppet_modules']), 1)
+
+        # Form environment name variable for our test
+        env_name = 'KT_{0}_{1}_{2}_{3}'.format(
+            org.name, lc_env.name, content_view.name, str(content_view.id))
+
+        # Get all environments for current organization.
+        # We have two environments (one created after publishing and one more
+        # was created after promotion), so we need to select promoted one
+        response = client.get(
+            entities.Environment().path(),
+            auth=get_server_credentials(),
+            data={
+                u'organization_id': org.id,
+            },
+            verify=False,
+        )
+        response.raise_for_status()
+        results = response.json()['results']
+        self.assertEqual(len(results), 2)
+        env_id = None
+        for result in results:
+            if result['name'] == env_name:
+                env_id = result['id']
+                break
+        environment = entities.Environment(
+            id=env_id
+        ).read()
+
+        # Prepare necessary data for HostGroup entity
+        mac = entity_fields.MACAddressField()
+        mac = mac.gen_value()
+        root_pass = entity_fields.StringField(length=(8, 30))
+        root_pass = root_pass.gen_value()
+        domain = entities.Domain().create(True)
+        architecture = entities.Architecture().create(True)
+        ptable = entities.PartitionTable().create(True)
+        operatingsystem = entities.OperatingSystem(
+            architecture=[architecture],
+            ptable=[ptable],
+        ).create(True)
+        medium = entities.Media(
+            operatingsystem=[operatingsystem]
+        ).create(True)
+
+        host_group = entities.HostGroup(
+            name=gen_string('alpha'),
+            architecture=architecture,
+            environment=environment,
+            operatingsystem=operatingsystem,
+            ptable=ptable,
+            domain=domain,
+            medium=medium,
+            location=[location.id],
+            organization=[org.id],
+        ).create()
+
+        self.assertEqual(len(host_group.read_json()['all_puppetclasses']), 0)
+
+        # Get puppet class id for ntp module
+        response = client.get(
+            entities.Environment(id=env_id).path()+'/puppetclasses',
```

Comment:
It makes sense to me that this path is listed under the "Puppetclasses" docs, not the "Environment" docs. `GET /api/environments/:environment_id/puppetclasses` returns puppet classes, so it should be listed under the "Puppetclasses" docs, right?

The fundamental problem here is that the `/api/environments/:environment_id/puppetclasses` path exists at all. From #88:

> Consider `GET /api/hosts/:host_id/puppetclass_ids`. It is one of
> three highly similar paths:
>
> 1. GET /api/puppetclasses
> 2. GET /api/hosts/:host_id/puppetclasses
> 3. GET /api/hosts/:host_id/puppetclass_ids
>
> The first call returns all puppet classes by default. You can filter down
> exactly how many puppet classes are returned by passing in a JSON payload such
> as `{'host_id': 42}`. The second call returns hashes of information about puppet
> classes that belong to a host. And the third call returns IDs of puppet classes
> that belong to a host.
>
> As you can see, the first API call is the most powerful, the second API call is
> less powerful and the third API call is least powerful. As a result, one can
> just use the first API call and ignore the other two, and no functionality is
> lost.

So yes, using this path works just fine. But a better long-term solution is to use this more powerful, general and orthogonal approach:

```python
response = client.get(
    entities.PuppetClass().path(),
    data={'environment_id': env_id},
    ...
)
```